lib/request.js

```diff
@@ -9,6 +10,12 @@ module.exports = (options) => {
     return new Promise((resolve, reject) => {
         const chunks = [];
         const request = httpModule.request(options, (response) => {
+            response.on('error', reject);
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response = response.pipe(zlib.createUnzip());
```

Comment:
Ah, good catch! Now it won't work because this function resolves with the `response`. I will fix it.